### DISCUSSION TITLE
Support launching iOS AR Quick Look from Facebook browser

### DISCUSSION THREAD

**Discussion by areavisuale** [google/model-viewer#1875] [Q&A]
## Description
If I open my webpage with model-viewer within the Facebook browser I cannot see the AR button. I tried directly in Chrome and Safari and all works. I hope this can be resolved soon

#### Live Demo
https://www.areavisuale.it/realta-aumentata.html

### OS

- [ ] Android
- [X] iOS

**Comment by cdata** [comment]
@areavisuale thanks for the report! We do not currently have a good way to reliably test the Facebook browser, as it requires a Facebook account to launch it. We could theoretically allow the button to show there, but I would stop short of guaranteeing support for it.

Do you happen to know the user agent that is presented by the Facebook browser, by any chance?

**Comment by areavisuale** [comment]
I tested the user agent using www.whatsmyua.info, and my iPhone X.

From Facebook the useragent is:

Mozilla/5.0 (iPhone; CPU iPhone OS 13_1_3 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/15E148 [FBAN/FBIOS;FBDV/iPhone10,6;FBMD/iPhone;FBSN/iOS;FBSV/13.1.3;FBSS/3;FBID/phone;FBLC/it_IT;FBOP/5;FBCR/Iliad]

From Chrome is:

Mozilla/5.0 (iPhone; CPU iPhone OS 13_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) CriOS/77.0.3865.103 Mobile/15E148 Safari/605.1

And from Safari is:

Mozilla/5.0 (iPhone; CPU iPhone OS 13_1_3 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/13.0.1 Mobile/15E148 Safari/604.1

Googling "Facebook Browser User Agent" someone advises to check for "FBAN" or "FBAV" and gives these examples:

Mozilla/5.0 (Linux; Android 7.1.1; Build/LMY47O.H18; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/58.0.3029.83 Mobile Safari/537.36 [FB_IAB/FB4A;FBAV/127.0.0.22.69;]
Mozilla/5.0 (Linux; Android 7.0; SM-G930F Build/NRD90M; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/58.0.3029.83 Mobile Safari/537.36 [FB_IAB/FB4A;FBAV/127.0.0.22.69;]
Mozilla/5.0 (Linux; Android 7.0; MHA-L29 Build/HUAWEIMHA-L29; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/58.0.3029.83 Mobile Safari/537.36 [FB_IAB/FB4A;FBAV/127.0.0.22.69;]
Mozilla/5.0 (iPhone; CPU iPhone OS 10_3_2 like Mac OS X) AppleWebKit/603.2.4 (KHTML, like Gecko) Mobile/14F89 [FBAN/FBIOS;FBAV/96.0.0.45.70;FBBV/60548545;FBDV/iPhone7,2;FBMD/iPhone;FBSN/iOS;FBSV/10.3.2;FBSS/2;FBCR/E-Plus;FBID/phone;FBLC/de_DE;FBOP/5;FBRV/0]
Mozilla/5.0 (iPhone; CPU iPhone OS 10_2 like Mac OS X) AppleWebKit/602.3.12 (KHTML, like Gecko) Mobile/14C92 [FBAN/FBIOS;FBAV/90.0.0.51.69;FBBV/56254015;FBDV/iPhone6,2;FBMD/iPhone;FBSN/iOS;FBSV/10.2;FBSS/2;FBCR/1&1;FBID/phone;FBLC/de_DE;FBOP/5;FBRV/0]
Mozilla/5.0 (iPhone; CPU iPhone OS 10_3_2 like Mac OS X) AppleWebKit/603.2.4 (KHTML, like Gecko) Mobile/14F89 [FBAN/FBIOS;FBAV/96.0.0.45.70;FBBV/60548545;FBDV/iPhone8,1;FBMD/iPhone;FBSN/iOS;FBSV/10.3.2;FBSS/2;FBCR/o2-de;FBID/phone;FBLC/de_DE;FBOP/5;FBRV/0]
Mozilla/5.0 (Linux; Android 4.4.4; G7-L01 Build/HuaweiG7-L01) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/33.0.0.0 Mobile Safari/537.36 [FB_IAB/MESSENGER;FBAV/121.0.0.15.70;]

**Comment by cdata** [comment]
Thanks @areavisuale that's great context

Could you also confirm for me that you can launch Quick Look from this test page in the Facebook browser: https://developer.apple.com/augmented-reality/quick-look/

**Comment by areavisuale** [comment]
Unfortunately not, the browser says "Impossible to open the file", "This type of file is not supported"
![IMG_2996](https://user-images.githubusercontent.com/39194720/68048543-1021c300-fce1-11e9-8081-002d26e0c9fe.jpg)

**Comment by cdata** [comment]
@areavisuale that's very interesting. That page is Apple's test page for launching AR Quick Look. What this suggests is that Facebook's built-in browser is not compatible with AR Quick Look, which means our AR button would not work even if we enabled it in Facebook's built-in browser.

Someday WebXR will allow us to activate AR in any web browser that supports it. But until then, we have to depend on browsers choosing to support proprietary native apps like iOS AR Quick Look and Google's Scene Viewer. And, there is no guarantee that a given browser will support these things, or if they do, that their support has the same set of requirements (this is why we have the `quick-look-browsers` attribute).

At this point, I'm afraid that there is not much we can do to better support the Facebook browser. But, I'm open to suggestions.
